data/src/main/scala/ch.epfl.scala.index.data/github/GithubDownload.scala

```diff
@@ -246,49 +257,98 @@ class GithubDownload(paths: DataPaths,
   }
 
   /**
-    * Process the downloaded data from GraphQL API
+    * Process the downloaded data from repository info
     *
     * @param repo the current repo
     * @param response the response
     * @return
     */
+  private def processCommunityProfileResponse(repo: GithubRepo,
+                                  response: WSResponse): Unit = {
+
+    if (200 == response.status) {
+
+      saveJson(githubRepoCommunityProfilePath(paths, repo), repo, response.body)
+    }
+
+    ()
+  }
+
+  /**
+    * Process the downloaded data from GraphQL API
+    * @param isRetry whether or not the call to this function is a retry
+    * @param repo the current repo
+    * @param response the response
+    * @return
+    */
   private def processTopicsResponse(isRetry: Boolean)(
       repo: GithubRepo,
-      response: WSResponse): Unit = {
+      response: WSResponse): Try[Unit] = Try {
+
+      checkGraphqlError[GithubRepo](repo, response, isRetry, topicQuery, processTopicsResponse)
+      saveJson(githubRepoTopicsPath(paths, repo), repo, response.body)
```

Comment:
Indeed I’m wrong, the `checkGraphqlError` still throws an exception. You may want to make it return a `Try[Unit]` instead.